### PR TITLE
Leaves - Hallie - Task-List

### PR DESCRIPTION
**Pull request by idhallie**
# Task List
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
Describe in your own words what the Model is doing in Rails | The model handles the database - stores and validates data.
Describe in your own words what the Controller is doing in Rails | The controller holds the tasks, which are like methods that give instructions upon receiving a request.
Describe in your own words what the View is doing in Rails | The View contains html/erb--this is where the browser output is designed.
Describe an edge-case controller test you wrote | Uhhhhhh...on the destroy action, after deleting the record, I look for the record by name to ensure that it's actually gone (because the id will persist). 
What is the purpose of using strong params? (i.e. the params method in the controller) | Strong params help keep your code DRY because you can define all the parameters in a method. More importantly, they help keep your data safe because you define the required and permitted fields and what data types they should take.
How are Rails migrations related to Rails models? | Whenever you alter your schema (ex. add or delete columns, change data types) you must create a migration and then run `rails db:migrate` for the change to take affect. 
Describe one area of Rails that are still unclear on | I'm still fuzzy on the model because we spent most of our time in the View and Controller. I understand what's happening in the database, but I think it would be helpful to spend more time learning what the Model can do.

### CODE REVIEW

File: test/controllers/tasks_controller_test.rb [hunk 1]
```diff
@@ -71,56 +70,109 @@
...
         post tasks_path, params: task_hash
       }.must_change "Task.count", 1
-
+      
       new_task = Task.find_by(name: task_hash[:task][:name])
       expect(new_task.description).must_equal task_hash[:task][:description]
-      expect(new_task.due_date.to_time.to_i).must_equal task_hash[:task][:due_date].to_i
-      expect(new_task.completed).must_equal task_hash[:task][:completed]
-
+      expect(new_task.completion_date).must_equal task_hash[:task][:completion_date]
```
Comment (dHelmgren) [inline test/controllers/tasks_controller_test.rb:81]: `DEPRECATED: Use assert_nil if expecting nil from test/controllers/tasks_controller_test.rb:81. This will fail in Minitest 6.`

File: test/controllers/tasks_controller_test.rb [hunk 2]
```diff
@@ -71,56 +70,109 @@
...
-
+  
   # Complete for Wave 4
   describe "toggle_complete" do
-    # Your tests go here
+    before do
+      @task2 = Task.create name: "test name", description: "test description"
+    end
+    
+    it "Can update a task from nil to completed and vice versa" do
```
Comment (dHelmgren) [inline test/controllers/tasks_controller_test.rb:166]: What if the ID is invalid?

File: config/routes.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+Rails.application.routes.draw do
+  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  root 'tasks#index'
+  
+  resources :tasks, except: [:update]
+  patch 'tasks/:id', to: 'tasks#update'
```
Comment (dHelmgren) [inline config/routes.rb:6]: This is a weird thing to do.

**Comment by dHelmgren** [comment]
# Task List
## What We're Looking For

Feature | Feedback
:------------- | :-------------
**Baseline** |
Appropriate Git Usage with no extraneous files checked in | yes
Answered comprehension questions | yes
Successfully handles: Index, Show | yes
Index & Show tests pass | yes
Successfully handles: New, Create | yes
New & Create tests pass | yes
Successfully handles: Edit, Update | yes
Tests for Edit & Update test valid & invalid task ids | yes
Successfully handles: Destroy, Task Complete | yes
Tests for Destroy & Task Complete include tests for valid and invalid task ids | see comments
Routes follow RESTful conventions | yes
Uses named routes (like `_path`) | yes
**Overall** | Good job on Task List! It functions as expected. You hit all of the learning goals: using partials, using strong params, and following Rails conventions.
